framework/CodeInterfaces/PHISICS/PhisicsRelapInterface.py

```diff
@@ -0,0 +1,257 @@
+  
+  def depTime(self,inputXML,searchDict,dictKeys):
+    """
+      Synchronizes the PHISICS time and RELAP time based on the input.
+      The input read are dep.xml and inst_cont.xml. The resulting ditionary is use in the "combine" class to print the PHISICS and RELAP
+      csv output in the proper time lines.
+      @ In, inputXML, depletion input name 
+      @ In, searchDict, dictionary, dictionary containing dummy keys, and the searched keywords as values
+      @ In, dictKeys, dictionary, dictionary containing dummy keys, the values are the keys of the output dict timeDict
+      @ Out, timeDict, dictionary
+    """
+    timeDict = {}
+    libraryTree = ET.parse(inputXML)
+    libraryRoot = libraryTree.getroot()
+    for key in searchDict:
+      for child in libraryRoot.getiterator(searchDict[key]):
+        timeDict[dictKeys[key]] = child.text
+        return timeDict
+
+  def _readMoreXML(self,xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class and initialize
+      some members based on inputs. This can be overloaded in specialize code interface in order to
+      read specific flags.
+      Only one option is possible. You can choose here, if multi-deck mode is activated, from which deck you want to load the results
+      @ In, xmlNode, xml.etree.ElementTree.Element, Xml element node
+      @ Out, None
+    """
+    self.xmlNode = xmlNode
+
+  def generateCommand(self,inputFiles,executable,clargs=None,fargs=None):
+    """
+      This method is used to retrieve the command (in tuple format) needed to launch the Code.
+      See base class.  Collects all the clargs and the executable to produce the command-line call.
+      Returns tuple of commands and base file name for run.
+      Commands are a list of tuples, indicating parallel/serial and the execution command to use.
+      @ In, inputFiles, list, List of input files (length of the list depends on the number of inputs have been added in the Step is running this code)
+      @ In, executable, string, executable name with absolute path (e.g. /home/path_to_executable/code.exe)
+      @ In, clargs, dict, optional, dictionary containing the command-line flags the user can specify in the input (e.g. under the node < Code >< clargstype =0 input0arg =0 i0extension =0 .inp0/ >< /Code >)
+      @ In, fargs, dict, optional, a dictionary containing the axuiliary input file variables the user can specify in the input (e.g. under the node < Code >< clargstype =0 input0arg =0 aux0extension =0 .aux0/ >< /Code >)
+      @ Out, returnCommand, tuple, tuple containing the generated command. returnCommand[0] is the command to run the code (string), returnCommand[1] is the name of the output root
+    """
+    dict = self.mapInputFileType(inputFiles)
+    outputfile = 'out~'+inputFiles[dict['relapInp'.lower()]].getBase()
+    self.outFileName = inputFiles[dict['relapInp'.lower()]].getBase()
```

Comment:
outFileName is in self because it is used in finalizeCodeOutput as well.

Reply:
why? ```outputfile``` is returned as input argument in the ```finalizeCodeOutput```. you do not store it in self because RAVEN returns it to you.

Fix please